apollo-ios-codegen/Sources/ApolloCodegenLib/Templates/SelectionSetTemplate.swift

```diff
@@ -538,13 +568,22 @@ struct SelectionSetTemplate {
     let isOptional =
       fragment.inclusionConditions != nil
       && !scope.matches(fragment.inclusionConditions.unsafelyUnwrapped)
+    let isDeferred = fragment.typeInfo.deferCondition != nil
 
     return """
```

Comment:
I think we might be at the point where it makes sense to break this function into smaller focused functions.